### PR TITLE
Add moderator command service

### LINKED ISSUE #63: Moderator Command Service
I'm not sure yet whether the moderator should be a separate aggregate, or a nested aggregate like in [here](https://docs.axoniq.io/reference-guide/axon-framework/axon-framework-commands/modeling/multi-entity-aggregates).

But we should provide a way to:
- [ ] Create / Asign Moderator
- [ ] Delete / Unasign Moderator

### PR DESCRIPTION
**Pull request by konradwojda**
Closes #63

### CODE REVIEW

File: shared/src/main/kotlin/pl/edu/pw/ia/shared/domain/command/ThreadCommands.kt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 package pl.edu.pw.ia.shared.domain.command
 
+import io.swagger.v3.oas.annotations.tags.Tag
```
Comment (bartlomiejkrawczyk) [inline shared/src/main/kotlin/pl/edu/pw/ia/shared/domain/command/ThreadCommands.kt:3]: I think that this is unused

File: threads/src/main/kotlin/pl/edu/pw/ia/threads/application/model/ModeratorsRequests.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+			subjectAccountId = subjectAccountId,
+			moderatorId = UUID.randomUUID(),
+			accountId = SecurityContext.getAccountId(),
+		)
+}
+
+data class RemoveModeratorRequest(
+
+	val threadId: UUID,
+	val subjectAccountId: UUID,
```
Comment (bartlomiejkrawczyk) [inline threads/src/main/kotlin/pl/edu/pw/ia/threads/application/model/ModeratorsRequests.kt:25]: We could leave here only subjectAccountId 
or leave a pair of threadId and accountId, and perform a query to lookup the subjectAccountId

File: threads/src/main/kotlin/pl/edu/pw/ia/threads/domain/command/aggregate/Moderator.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+				threadId = command.threadId,
+				accountId = command.subjectAccountId
+			)
+		)
+	}
+
+	@CommandHandler
+	fun handle(command: RemoveModeratorCommand) {
+		// TODO: Check if command.accountId is moderator in command.threadId
+		// TODO: Should we allow moderator to delete himself from moderators?
```
Comment (bartlomiejkrawczyk) [inline threads/src/main/kotlin/pl/edu/pw/ia/threads/domain/command/aggregate/Moderator.kt:38]: hmm, we shouldn't allow to delete only the original moderator - aka post creator 🤔

**Review by bartlomiejkrawczyk — APPROVED** [review]
only minor modifications are required

File: threads/src/main/kotlin/pl/edu/pw/ia/threads/application/model/ModeratorsRequests.kt (hunk 2 above)
Comment (konradwojda) [reply threads/src/main/kotlin/pl/edu/pw/ia/threads/application/model/ModeratorsRequests.kt]: Changed to `moderatorId`

File: threads/src/main/kotlin/pl/edu/pw/ia/threads/domain/command/aggregate/Moderator.kt (hunk 3 above)
Comment (konradwojda) [reply threads/src/main/kotlin/pl/edu/pw/ia/threads/domain/command/aggregate/Moderator.kt]: Let's assume only author of the thread can modify moderators.

File: shared/src/main/kotlin/pl/edu/pw/ia/shared/domain/event/ThreadEvents.kt [hunk 4]
```diff
@@ -29,14 +29,14 @@ data class ThreadDeleteEvent(
...
 @Revision("1.0")
 data class ModeratorAddedEvent(
+	val moderatorId: UUID,
 	val threadId: UUID,
 	val accountId: UUID,
 )
 
 @Revision("1.0")
 data class ModeratorRemovedEvent(
-	val threadId: UUID,
```
Comment (bartlomiejkrawczyk) [inline shared/src/main/kotlin/pl/edu/pw/ia/shared/domain/event/ThreadEvents.kt:38]: we could leave this here

File: threads/src/main/kotlin/pl/edu/pw/ia/threads/application/model/ModeratorsRequests.kt [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+			accountId = SecurityContext.getAccountId(),
+		)
+}
+
+data class RemoveModeratorRequest(
+
+	val moderatorId: UUID,
+) {
+	fun toCommand(): RemoveModeratorCommand =
+		RemoveModeratorCommand(
```
Comment (bartlomiejkrawczyk) [inline threads/src/main/kotlin/pl/edu/pw/ia/threads/application/model/ModeratorsRequests.kt:27]:
```
accountId = SecurityContext.getAccountId(),
```

**Pull request merged**